### ISSUE TITLE
Calling convention run-time check failure

### ISSUE THREAD

**Issue by abadd** [cginternals/glbinding#42]
The following program encounters a run-time failure:

```
#include <SFML\Window.hpp>
#include <glbinding\gl\gl.h>
#include <glbinding\Binding.h>

using namespace gl;

int main(int argc, const char* argv[])
{
    sf::Window window(sf::VideoMode(640, 480), "GLBinding Test");

    glbinding::Binding::initialize();
    glClearColor(0.0f, 0.5f, 0.8f, 1.0f);
   // truncated
}
```
With the error message:

> Run-Time Check Failure #0 - The value of ESP was not properly saved across a function call.  This is usually a result of calling a function declared with one calling convention with a function pointer declared with a different calling convention.

Which occurs on line 71 of Function.hpp and the call stack for the error originates from the `glClearColor` call:

```
template <typename... Arguments>
struct FunctionHelper<void, Arguments...>
{
    void call(const glbinding::Function<void, Arguments...> * function, Arguments&&... arguments) const
    {
        // truncated
        if (function->isEnabled(glbinding::CallbackMask::After))      // line 71
            function->after(functionCall);
    }
};
```

The toolchain used to build the library:
* VS 2013 Version 12.0.30723.00 Update 3
* CMake 3.0.2

The same tools were used to build SFML and GLEW which have no issues with the same program.

**Comment by imallett** [comment]
I can confirm this is still broken on x86. x86-64 build is fine.

**Comment by scheibel** [comment]
We currently tried to reproduce this error on our windows machines and it didn't pop up.
This issue is over one year old; can anyone confirm the issue is still there in the current master?

**Comment by zao** [comment]
I re-validated my claims on VS2015.1 x86:
The commit before the ac049cc2d8ec9de837f496ff569a8bbc180c499b fix (7f063b2fe6baaaf03ba505132d8969bddaf09a5) has the mismatched stack diagnostic as reported.
A build of ac049cc2d8ec9de837f496ff569a8bbc180c499b does not exhibit the reported problem.
Current master (b838090719a72207c020025b24a26fa9cf34fbbe) does not exhibit the reported problem.

**Comment by cgcostume** [comment]
Currently we don't plan to fix this issue in glbinding 1.1 (c031b7f0830146519dad9451a60a764c41cacfb8) branch and since it cannot be reproduced for 2.0 the issue will be closed for now (@julkw tested x86 builds with Qt, GLFW, and GLEW on Windows). However, should the issue emerge again, especially for SFML, please re-open.
